File: automation-portal/src/app/device/device.ts
import { CustomFieldValue } from "../custom-field/custom-field-value";
import { Site } from "../site/site";

export class Device {
    id: number = 0;
    name: string = '';
    address: string = '';
    deviceTypeId: number = 0;
    monitoringId: string = '';
    enabled: boolean = true;
    siteId: number | undefined | null;
    site: Site = new Site();
    customFieldValue: CustomFieldValue[] = new Array<CustomFieldValue>();
}

File: api/AutomationPortal/DB/Entity/Device.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Threading.Tasks;

namespace AutomationPortal.DB.Entity
{
    public class Device
    {
        public int Id { get; set; }
        [Required]
        public string Name { get; set; }
        [Required]
        public string Address { get; set; }
        public string MonitoringId { get; set; }
        public bool Enabled { get; set; }
        public int SiteId { get; set; }
        public Site Site { get; set; }

        public int DeviceTypeId { get; set; }
        public DeviceType DeviceType { get; set; }
        
        public ICollection<DeviceCustomFieldValue> CustomFieldValue { get; set; }
    }
}
